proxy/src/usage_metrics.rs

```diff
@@ -213,25 +337,141 @@ async fn collect_metrics_iteration(
 
         if !res.status().is_success() {
             error!("metrics endpoint refused the sent metrics: {:?}", res);
-            for metric in chunk.iter().filter(|(_, value)| *value > (1u64 << 40)) {
+            for metric in chunk.events.iter().filter(|e| e.value > (1u64 << 40)) {
                 // Report if the metric value is suspiciously large
                 error!("potentially abnormal metric value: {:?}", metric);
             }
         }
     }
+}
 
-    for metric in metrics_to_clear {
-        match metrics.endpoints.entry(metric) {
-            Entry::Occupied(mut counter) => {
-                if counter.get_mut().should_clear() {
-                    counter.remove_entry();
-                }
+pub async fn task_backup(
```

Comment:
Do we need separate a separate task here? I think it would make more sense to upload to S3 in the same collection task from above. More importantly, I feel like the `collect_and_clear_metrics` being run from two tasks will introduce race conditions where it's cleared by one task and the other does not observe it.

Reply:
Also, sorry for this code. I am not so proud of it, but it solved it's purpose for the time and it was no longer ruining the memory of the process so I didn't want to think of it further 😓

Reply:
> Do we need separate a separate task here?

i guess so, because they have different intervals, also uploads everything to the different remote storages.

No, it shouldn't be the case. One touches `endpoints` and upload it to `vector`. And the other one touches just `backup_endpoints`.